portfolio-project-react/src/components/Details/Details.tsx

```diff
@@ -1,33 +1,46 @@
-import { useState, useEffect, useMemo } from "react";
-import About from "../About/About";
-import WorkExperience from "../WorkExperience/WorkExperience";
+import React, { useState, useEffect, useMemo } from "react";
+import About from "../About/About.tsx";
+import WorkExperience from "../WorkExperience/WorkExperience.tsx";
 import styles from "./Details.module.css";
-import { WORK_EXPERIENCE_ITEM_ADDED } from "./constants";
+import { WORK_EXPERIENCE_ITEM_ADDED } from "./constants.tsx";
 
+interface WorkExperienceProps {
+  _id:string;
+  companyName: string;
+  startDate: any;
+  endDate: any;
+  description: string;
+}
+interface NoOfDays {
```

Comment:
you have declared the same kind of interface in About.tsx. Please extract common interfaces and create a file; interface.tsx where all the commonly used and generic types should be placed. 

You'll need to check in the rest of the places as well.